extensions/amp-story/1.0/amp-story-affiliate-link.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Affiliate link component that expands when clicked.
+ */
+
+import {Services} from '../../../src/services';
+import {StateProperty, getStoreService} from './amp-story-store-service';
+import {getAmpdoc} from '../../../src/service';
+import {htmlFor} from '../../../src/static-template';
+
+/**
+ * Links that are affiliate links.
+ * @const {string}
+ */
+export const AFFILIATE_LINK_SELECTOR = 'a[affiliate-link-icon]';
+
+/**
+ * Custom property signifying a built link.
+ * @const {string}
+ */
+export const AFFILIATE_LINK_BUILT = '__AMP_AFFILIATE_LINK_BUILT';
+
+export class AmpStoryAffiliateLink {
+  /**
+   * @param {!Window} win
+   * @param {!Element} element
+   */
+  constructor(win, element) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {!Element} */
+    this.element_ = element;
+
+    /** @private {?Element} */
+    this.textEl_ = null;
+
+    /** @private {?Element} */
+    this.launchEl_ = null;
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(this.win_);
+
+    /** @private {string} */
+    this.text_ = this.element_.textContent;
+
+    /** @private @const {!../../../src/service/resources-impl.ResourcesDef} */
+    this.resources_ = Services.resourcesForDoc(getAmpdoc(this.win_.document));
+  }
+
+  /**
+   * Builds affiliate link.
+   */
+  build() {
+    if (this.element_[AFFILIATE_LINK_BUILT]) {
+      return;
+    }
+
+    this.resources_.mutateElement(this.element_, () => {
+      this.element_.textContent = '';
```

Comment:
Should this be `this.element_.innerHTML = ''`?  What if I put, for example, a cat gif inside of my affiliate link?

Reply:
The cat gif gets stripped out when getting the text, which I think should be intended?

Reply:
Oh, yeah, that's intended, I just didn't know that's how this worked.  Thanks for clarifying!

Reply:
:+1: should I merge?

Reply:
Can you track the next steps somewhere as you merge this PR? Especially RTL support, visual tests, and validation :)
Thanks!

Reply:
Sure thing, updated the issue https://github.com/ampproject/amphtml/issues/22703